main.go

```diff
@@ -73,26 +61,27 @@ func main() {
 		},
 	}
 
-	// If the user has changed the port via environment variable, respect this.
-	var portToUse string
-	if *port == 8080 {
-		portToUse = os.Getenv("PROXY_APP_PORT")
-		if portToUse == "" {
-			portToUse = fmt.Sprintf("%d", *port)
-		}
-	} else {
-		portToUse = fmt.Sprintf("%d", *port)
-	}
-
 	// Debug messaging.
-	fmt.Printf("Proxy configured to use port %s\n", portToUse)
+	fmt.Printf("Proxy configured to use port %s\n", PROXY_APP_PORT)
 	for _, target := range targets {
-		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", portToUse, target.URL)
+		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", PROXY_APP_PORT, target.URL)
+	}
+
+	// Handle Authentication
+	if PROXY_APP_USERNAME != "" && PROXY_APP_PASSWORD != "" {
+		e.Use(middleware.BasicAuth(func(username, password string, c echo.Context) (bool, error) {
```

Comment:
Looks like this is setting up basic auth for the "client -> api" and not for the reverse proxy connection to the backend.

Reply:
Removed.